src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -612,6 +612,109 @@ describe('Transactions Controller (Unit)', () => {
         });
     });
   });
+
+  describe('GET queued transactions by Safe', () => {
+    it('should get a transactions queue with labels and conflict headers', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = new ChainBuilder().withChainId(chainId).build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrl = `${safeConfigApiUrl}/api/v1/chains/${chainId}`;
+        const getMultisigTransactionsUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/multisig-transactions/`;
+        const getSafeUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}`;
+        const getContractUrlPattern = `${chainResponse.transactionService}/api/v1/contracts/`;
+        if (url === getChainUrl) {
+          return Promise.resolve({ data: chainResponse });
+        }
+        if (url === getMultisigTransactionsUrl) {
+          return Promise.resolve({
+            data: getJsonResource(
+              'queued-items/single-page/transactions-source-data.json',
+            ),
+          });
+        }
+        if (url === getSafeUrl) {
+          return Promise.resolve({
+            data: getJsonResource(
+              'queued-items/single-page/safe-source-data.json',
+            ),
+          });
+        }
+        if (url.includes(getContractUrlPattern)) {
+          return Promise.resolve({
+            data: getJsonResource(
+              'queued-items/single-page/contract-source-data.json',
+            ),
+          });
+        }
+        return Promise.reject(new Error(`Could not match ${url}`));
+      });
```

Comment:
Other conditions don't get tested because the first matching condition will exit the function calling `return`. But I'm not against using the `if else` notation if you prefer, or if you find it clearer 🙂

Reply:
Ah yes you are right. I missed the `return` statement 😬